components/HorizontalBarChart.vue

```diff
@@ -0,0 +1,153 @@
+      :chart-data="displayData"
+      :options="displayOption"
+      :height="240"
+      :plugins="[getPlugins()]"
+    />
+    <div v-if="hasNoData()" class="no-data">
+      No Data Avail
+    </div>
+  </data-view>
+</template>
+<script>
+import ChartDataLabels from 'chartjs-plugin-datalabels'
+import DataView from '@/components/DataView.vue'
+import { ChartTypes } from '@/utils/formatCountyDataVTwo'
+export default {
+  components: { DataView },
+  props: {
+    title: {
+      type: String,
+      required: false,
+      default: ''
+    },
+    titleId: {
+      type: String,
+      required: false,
+      default: ''
+    },
+    chartId: {
+      type: String,
+      required: false,
+      default: 'horizontal-bar-chart'
+    },
+    chartData: {
+      type: Object,
+      required: false,
+      default: () => {}
+    },
+    date: {
+      type: String,
+      required: true,
+      default: ''
+    },
+    url: {
+      type: String,
+      required: false,
+      default: ''
+    }
+  },
+  computed: {
+    displayData() {
+      return {
+        labels: this.chartData.labels,
+        datasets: this.chartData.datasets
+      }
+    },
+    displayOption() {
+      return {
+        tooltips: {
+          displayColors: false,
+          showAllTooltips: true,
+          titleAlign: 'center',
+          bodyAlign: 'center',
+          callbacks: {
+            label(tooltipItem) {
+              if (!tooltipItem) return
+              return 'cases'
+            },
+            title(tooltipItem) {
+              if (!tooltipItem) return
+              return tooltipItem[0].value
+            }
+          }
+        },
+        barValueSpacing: 20,
+        responsive: true,
+        maintainAspectRatio: false,
+        legend: {
+          display: this.chartData.displayLegend
+        },
+        scales: {
+          xAxes: [
+            {
+              id: 'cases',
+              stacked: false,
+              gridLines: {
+                display: true,
+                drawOnChartArea: false
+              },
+              ticks: {
+                fontSize: 9,
+                maxTicksLimit: 30,
+                fontColor: '#808080',
+                maxRotation: 0,
+                minRotation: 0
+              }
+            }
+          ],
+          yAxes: [
+            {
+              stacked: false,
+              gridLines: {
+                display: true,
+                color: '#E5E5E5',
+                drawOnChartArea: false
+              },
+              ticks: {
+                suggestedMin: 0,
+                maxTicksLimit: 8,
+                fontColor: '#808080'
+              }
+            }
+          ]
+        },
+        plugins: this.chartData.customChartOptions.plugins
+      }
+    }
+  },
+  methods: {
+    hasNoData() {
+      return this.chartData.datasets.length < 1
+    },
+    getPlugins() {
+      if (this.chartData.chartType === ChartTypes.GENDER) return ChartDataLabels
```

Comment:
minor comment tho, I would make the code consistent with other cases.

Reply:
@kengoy what do you mean?

Reply:
I just meant only this if condition does not break a line of the "return chartDataLabels" but other two if cases do. Just for the code readiness perspective to make it the same policy. It's not a big deal.

I just noticed now the return value is all same among the 3 if conditions? I would make it one if condition.

Reply:
oh you're right lol. Thanks for the catch!

Reply:
@kengoy Oh I remember now why I listed it with multiple ChartTypes. The plugin ChartDataLabels is an extra feature that exists only on certain chart types. This plugin allows the display of the tooltip all the time. The getPlugins is my workaround for enabling it for certain horizontal charts. In my case, I only want it for Sex, Race & Ethn, and Race & Ethn norm. Excluding it from Age chart. 

I listed it in this order because in the future, if we wanted to add more horizontal bar charts with the tooltips displayed, you can just add the ChartType check to the list.